pkg/either/either.go

```diff
@@ -0,0 +1,43 @@
+package either
+
+// Either represents a type that can hold either a value of type T or an error.
+// It's commonly used in functional programming to encapsulate functions that might fail.
+// This way, instead of returning a value and an error separately, both are combined into a single type.
+type Either[R any] struct {
```

Comment:
What is the thinking behind Not having an `EitherErr`?

It feels kind of weird that EIther is not a generic `R, L` with a helper to define a subset called `R, error`

Reply:
Because `L` would always be of `error` type. We would be adding a superfluous generic type describing it.

Also, if we go with the `EitherErr`, the definition in the comments abobe `Either` struct would apply to `EitherErr` and not `Either` which makes `EitherErr` be the `Either` itself (by definition).